core/src/main/java/org/geysermc/geyser/GeyserBootstrap.java

```diff
@@ -44,14 +44,24 @@ public interface GeyserBootstrap {
     GeyserWorldManager DEFAULT_CHUNK_MANAGER = new GeyserWorldManager();
 
     /**
-     * Called when the GeyserBootstrap is enabled
+     * Called when the GeyserBootstrap is initialized
      */
-    void onEnable();
+    void onGeyserInitialize();
+
+    /**
+     * Called when the GeyserBootstrap is enabled/reloaded
+     */
+    void onGeyserEnable();
 
     /**
      * Called when the GeyserBootstrap is disabled
      */
-    void onDisable();
+    void onGeyserDisable();
+
+    /**
+     * Called when the GeyserBootstrap is shutting down.
+     */
+    void onGeyserShutdown();
```

Comment:
Can we be more detailed here - doesn't have to be as specific as this PR description but more nuance would be nice